libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -140,6 +204,8 @@ string IRGenerator::generate(
 	generateImplicitConstructors(_contract);
 	generateQueuedFunctions();
 	InternalDispatchMap internalDispatchMap = generateInternalDispatchFunctions();
+
+	m_creationFunctionList = m_context.functionCollector().requestedFunctionsNames();
```

Comment:
Is that the right way to collect them? Would it be better to do this in `IRGenerator::generateFunction(FunctionDefinition`?

Reply:
I mean we could collect AST nodes instead of IR function names - wouldn't that be more precise?